lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -612,5 +612,162 @@ def test_laea_cs(self):
         self._test(coord_system, expected)
 
 
+@tests.skip_data
+class Test_packed_data(tests.IrisTest):
+    def _get_scale_factor_add_offset(self, cube, datatype):
+        dt = np.dtype(datatype)
+        cmax = cube.data.max()
+        cmin = cube.data.min()
+        n = dt.itemsize * 8
+        if isinstance(cube.data, np.ma.core.MaskedArray):
+            masked = True
+        else:
+            masked = False
+        if masked:
+            scale_factor = (cmax - cmin)/(2**n-2)
+        else:
+            scale_factor = (cmax - cmin)/(2**n-1)
+        if dt.kind == 'u':
+            add_offset = cmin
+        elif dt.kind == 'i':
+            if masked:
+                add_offset = (cmax + cmin)/2
+            else:
+                add_offset = cmin + 2**(n-1)*scale_factor
+        return (scale_factor, add_offset)
+
+    def test_single_packed_signed(self):
+        """Test saving a single CF-netCDF file with packing."""
+        # Read PP input file.
+        file_in = tests.get_data_path(
+            ('PP', 'cf_processing',
+             '000003000000.03.236.000128.1990.12.01.00.00.b.pp'))
+        cube = iris.load_cube(file_in)
```

Comment:
@jswanljung `L642:645` common to all `test_single_packed_...` type tests, so you could put that in a `def setUp(self)`.